examples/chip-tool/commands/reporting/ReportingCommand.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ *   Copyright (c) 2020 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "ReportingCommand.h"
+
+#include "../common/Commands.h"
+#include "gen/CHIPClientCallbacks.h"
+#include <controller/CHIPClusters.h>
+
+using namespace ::chip;
+
+namespace {
+constexpr uint16_t kWaitDurationInSeconds = UINT16_MAX;
+} // namespace
+
+CHIP_ERROR ReportingCommand::Run(PersistentStorage & storage, NodeId localId, NodeId remoteId)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::Controller::BasicCluster cluster;
+
+    err = mCommissioner.SetUdpListenPort(storage.GetListenPort());
+    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Init failure! Commissioner: %s", ErrorStr(err)));
```

Comment:
nit: I would suggest a different string as it's the same as the one below.  It could make it difficult to determine if the failure is with the UDP port or the initialization